devito/__init__.py

```diff
@@ -48,9 +48,9 @@ def _reinit_compiler(val):  # noqa
     # Force re-build the compiler
     configuration['compiler'].__init__(suffix=configuration['compiler'].suffix,
                                        mpi=configuration['mpi'])
-    return bool(val)
+    return bool(val) if isinstance(val, int) else val
 configuration.add('openmp', 0, [0, 1], callback=_reinit_compiler)  # noqa
-configuration.add('mpi', 0, [0, 1], callback=_reinit_compiler)
+configuration.add('mpi', 0, [0, 1, 'basic', 'diag'], callback=_reinit_compiler)
```

Comment:
what are these new two? Is it documented?

Reply:
undocumented. Basically, currently, when you set it to 1 you get 'basic'. This is heavily work in progress and likely to be changed drastically as soon as computation/communication overlap lands